python/pydantic_core/core_schema.py

```diff
@@ -1384,6 +1384,24 @@ def uuid_schema(
     )
 
 
+class NestedModelSchema(TypedDict, total=False):
```

Comment:
I wonder if we'd want to just use something like `RefSchema` or something like that so that we can extend this beyond `BaseModel`

Reply:
definitely